tests/cxx/iwyu_stricter_than_cpp-i4.h

```diff
@@ -0,0 +1,16 @@
+//===--- iwyu_stricter_than_cpp-i4.h - test input file for iwyu -----------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+class IndirectClass;
+
+struct IndirectStruct4 {
+  typedef IndirectClass IndirectClassNonprovidingTypedef;
```

Comment:
Could you use `NonProviding` instead of `Nonproviding` throughout?

Reply:
Of course, if you insist. But [here](https://github.com/include-what-you-use/include-what-you-use/blob/0.18/iwyu_ast_util.h#L619) `Nonclass` spelling is used.

Reply:
Yeah, I was hoping we could clean that up as part of renaming to `Tag` :-)